daml-lf/archive/src/main/scala/com/digitalasset/daml/lf/archive/UniversalArchiveReader.scala

```diff
@@ -34,7 +34,9 @@ class UniversalArchiveReader[A](
   /** Reads a DAR from a File. */
   def readFile(file: File): Try[Dar[A]] =
     supportedFileType(file).flatMap {
-      case DarFile => readDarStream(file.getName, new ZipInputStream(new FileInputStream(file)))
+      case DarFile =>
```

Comment:
The resource owner ship here is pretty messy. `readDarStream` and  `readDalfStream` both close an input stream passed in to them. IMHO you should not close resources that you didn’t allocate but that’s independent of the issue here.